src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -172,4 +179,42 @@ private void validateColumnForPartition(@Nullable String columnName, @Nullable S
                                                     columnName, type));
     }
   }
+
+  private void validateClusteringOrder(@Nullable Schema schema) {
+    if (!shouldCreatePartitionedTable() || Strings.isNullOrEmpty(clusteringOrder) || schema == null) {
+      return;
+    }
+    List<String> columnsNames = Arrays.stream(clusteringOrder.split(",")).map(String::trim)
+      .collect(Collectors.toList());
+    if (columnsNames.size() > MAX_NUMBER_OF_COLUMNS) {
+      throw new InvalidConfigPropertyException(
+        String.format("Expected no more than '%d' clustering fields, found '%d'.", MAX_NUMBER_OF_COLUMNS,
+                      columnsNames.size()), NAME_CLUSTERING_ORDER);
+    }
+    for (String column : columnsNames) {
+      Schema.Field field = schema.getField(column);
+      if (field == null) {
+        throw new InvalidConfigPropertyException(
+          String.format("Clustering column '%s' is missing from the table schema", column),
+          NAME_CLUSTERING_ORDER);
+      }
+      Schema.Type type = field.getSchema().isNullable()
+        ? field.getSchema().getNonNullable().getType()
+        : field.getSchema().getType();
+      Schema.LogicalType logicalType = field.getSchema().isNullable()
+        ? field.getSchema().getNonNullable().getLogicalType()
+        : field.getSchema().getLogicalType();
+      if (!BigQueryUtil.SUPPORTED_CLUSTERING_TYPES.contains(type)) {
+        throw new InvalidConfigPropertyException(
+          String.format("Field '%s' has type '%s', which is not supported for clustering.", column, type),
+          NAME_CLUSTERING_ORDER);
+      }
+      if (logicalType != Schema.LogicalType.DATE && logicalType != Schema.LogicalType.TIMESTAMP_MICROS
+        && logicalType != Schema.LogicalType.TIMESTAMP_MILLIS && logicalType !=  Schema.LogicalType.DECIMAL) {
```

Comment:
remove extra space in front of Schema.LogicalType.DECIMAL

Reply:
Done.